### PR TITLE
Code tidy-up

### PR DESCRIPTION
**Pull request by dladd**
Resolves a few small differences in style outlined in issue #75

### CODE REVIEW

**Comment by dladd** [comment]
I believe this now covers the items specified in issue #75 . 

You can find a link to the built [Coding Standard page here](http://autotest.bioeng.auckland.ac.nz/libcellml-buildbot/0e4b52cbcc1f0bcd15258d2ef2cf09b763c253ff/dox/dev_coding_standard.html) for convenience.

File: docs/dev_coding_standard.rst [hunk 1]
```diff
@@ -10,6 +10,20 @@ In essence we seek code that looks good, is easy to read and has great documenta
...
 
  * lower camel case class method names
+ * indent code 4 spaces at a time (no tabs)
+
+Doxygen Comments
+================
+
+ * Doxygen comments should be sentence-style: beginning with capitalisation and ending with punctuation. However, they do not need to form grammatically correct sentences.
+
+   * An exception for capitalisation is code-words (e.g., true, false, std::string), which should be styled as typewriter text with a preceding "@c".
```
Comment (agarny) [inline docs/dev_coding_standard.rst]: Shouldn't class names also be styled as typewriter (e.g., `Entity`, `Units`)?

File: docs/dev_coding_standard.rst [hunk 2]
```diff
@@ -10,6 +10,20 @@ In essence we seek code that looks good, is easy to read and has great documenta
...
+ 
+Test Naming
+===========
+
+ * lower camel case
```
Comment (agarny) [inline docs/dev_coding_standard.rst]: Replace with "Names should respect the lower camel case convention." or something along those lines.

File: src/api/libcellml/importedentity.h [hunk 3]
```diff
@@ -42,15 +42,15 @@ class LIBCELLML_EXPORT ImportedEntity: public NamedEntity
      * @brief Test if this entity is an imported entity.
      * Method to test if this entity is an imported entity.  Returns
      * true if the import is not the nullptr otherwise return false.
-     * @return true if the import is not the nullptr, false otherwise.
+     * @return @c true if the import is not the nullptr, @c false otherwise.
```
Comment (agarny) [inline src/api/libcellml/importedentity.h]: Replace "the nullptr" with "@c nullptr".

File: src/api/libcellml/importedentity.h [hunk 4]
```diff
@@ -42,15 +42,15 @@ class LIBCELLML_EXPORT ImportedEntity: public NamedEntity
...
      */
     bool isImport() const;
 
     /**
      * @brief Get the import set.
      * Get the import set, if no import is set nullptr is returned.
```
Comment (agarny) [inline src/api/libcellml/importedentity.h]: Replace "nullptr" with "@c nullptr".

File: src/api/libcellml/importedentity.h [hunk 5]
```diff
@@ -42,15 +42,15 @@ class LIBCELLML_EXPORT ImportedEntity: public NamedEntity
...
      * @sa setImport
-     * @return the shared pointer for the import, if no import is set returns nullptr.
+     * @return The shared pointer for the import, if no import is set returns nullptr.
```
Comment (agarny) [inline src/api/libcellml/importedentity.h]: Replace "nullptr" with "@c nullptr".

File: src/api/libcellml/namedentity.h [hunk 6]
```diff
@@ -51,7 +51,7 @@ class LIBCELLML_EXPORT NamedEntity: public Entity
     /**
      * @brief Get the name of the Entity
      * Gets the name of the Entity as a string.
-     * @return std::string representation of the Entity name.
+     * @return @c std::string representation of the Entity name.
```
Comment (agarny) [inline src/api/libcellml/namedentity.h:54]: Replace "Entity name" with "@c Entity name"?

File: src/api/libcellml/units.h [hunk 7]
```diff
@@ -87,15 +87,15 @@ class LIBCELLML_EXPORT Units: public ImportedEntity
      * @brief Test to determine if Units is a base unit.
      * Test to determine if Units is a base unit, return True if it is
      * a base unit and False otherwise.
-     * @return True if Units is a base unit, False otherwise.
+     * @return @c true if Units is a base unit, @c false otherwise.
```
Comment (agarny) [inline src/api/libcellml/units.h:90]: Replace "Units" with "@c Units"?

File: src/model.cpp [hunk 8]
```diff
@@ -120,10 +120,10 @@ std::string Model::doSerialisation(libcellml::FORMATS format) const
         }
         repr += ">";
 
-        for (ImportMapIterator iter = importMap.begin(); iter != importMap.end(); iter++)
+        for (ImportMapIterator iter = importMap.begin(); iter != importMap.end(); ++iter)
         {
             repr += "<import xlink:href=\"" + iter->first->getSource() + "\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">";
-            for (VectorPairIterator vector_iter = iter->second.begin(); vector_iter != iter->second.end(); vector_iter++) {
+            for (VectorPairIterator vector_iter = iter->second.begin(); vector_iter != iter->second.end(); ++vector_iter) {
```
Comment (agarny) [inline src/model.cpp]: Replace `vector_iter` with `vectorIter`.

**Comment by agarny** [comment]
It all looks good, pending a few minor issues and my comment [here](https://github.com/cellml/libcellml/issues/75#issuecomment-209826989) about 2-space _vs_ 4-space indentation in CMake _vs_ C++ files.

File: docs/dev_coding_standard.rst (hunk 1 above)
Comment (dladd) [inline docs/dev_coding_standard.rst]: Actually, Doxygen picks up class names as long as they are capitalised and links back to the relevant class in the generated documentation. For instance, [here's the generated version](http://autotest.bioeng.auckland.ac.nz/libcellml-buildbot/b8a163fb017ccc9def0bd1eea60b7f25f9020705/dox/code/classlibcellml_1_1NamedEntity.html#a21ef5ac512cea802de725a0a729cf970) of the Entity class mention in the getName function you referred to below.

**Comment by dladd** [comment]
Thanks, @agarny , I've added changes per your comments.

File: docs/dev_coding_standard.rst (hunk 1 above)
Comment (agarny) [inline docs/dev_coding_standard.rst]: Oops, my bad!

File: docs/dev_coding_standard.rst [hunk 9]
```diff
@@ -10,19 +10,18 @@ In essence we seek code that looks good, is easy to read and has great documenta
...
-
-   * An exception for capitalisation is code-words (e.g., true, false, std::string), which should be styled as typewriter text with a preceding "@c".
+ * Code-words (e.g., true/false, std::string) should be styled as typewriter text with a preceding "@c".
+ * Doxygen comments should be sentence-style: beginning with capitalisation (except code-words) and ending with punctuation. However, they do not need to form grammatically correct sentences.
  
 Test Naming
 ===========
 
- * lower camel case 
+ * Names should respect lower camel case convention.
```
Comment (agarny) [inline docs/dev_coding_standard.rst]: "... respect [the] lower camel..."?

**Comment by agarny** [comment]
Besides my very minor comment above, it all looks fine to me and I wouldn't mind being merged _as is_ (i.e. without taking care of my latest comment).

**Comment by dladd** [comment]
Thanks @agarny , I've just addressed that grammatical error.

**Comment by agarny** [comment]
Oops, forgot to mention that the merge effectively closes #75.

**Pull request merged**